crates/astria-conductor/src/conductor.rs

```diff
@@ -174,19 +187,29 @@ impl Conductor {
     async fn run_until_stopped(mut self) {
         info_span!("Conductor::run_until_stopped").in_scope(|| info!("conductor is running"));
 
-        let exit_reason = select! {
-            biased;
+        let exit_reason = loop {
+            select! {
+                biased;
 
-            () = self.shutdown.cancelled() => {
-                Ok("received shutdown signal")
-            }
+                () = self.shutdown.cancelled() => {
+                    break Ok("received shutdown signal");
+                }
 
-            Some((name, res)) = self.tasks.join_next() => {
-                match flatten(res) {
-                    Ok(()) => Err(eyre!("task `{name}` exited unexpectedly")),
-                    Err(err) => Err(err).wrap_err_with(|| "task `{name}` failed"),
+                Some((name, res)) = self.tasks.join_next() => {
+                    match flatten(res) {
+                        Ok(()) => break Err(eyre!("task `{name}` exited unexpectedly")),
+                        Err(err) => {
+                            if check_for_restart(&err) {
```

Comment:
The assumption here is that this works if the executor-task exits first, followed by the other tasks.

There is no guarantee however for that to happen: if executor has lengthy cleanup code after it exits its event loop while other the other tasks exit immediately, then this check is guaranteed to no longer work.

Reply:
Added checks for the `PermissionDenied` code for the remaining tasks in `shutdown()` in d6fde21.